src/components/WebChat/WebChat.js

```diff
@@ -1,80 +1,95 @@
-import React, { useState, useEffect, useRef, useMemo } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import { BsX, BsCursorFill } from "react-icons/bs";
 import { messageHistoryGenerator } from "./MessageGeneratorUtil";
 
-import "./WebChat.css";
-
 const history = {
   messages: messageHistoryGenerator(),
 };
 
 export default function WebChat() {
+ 
   const [chatHistory, setChatHistory] = useState(history.messages);
   const [isChatBoxVisible, setIsChatBoxVisible] = useState(false);
+  const [message, setMessage] = useState('');
   const chatBox = useRef(null);
 
   useEffect(() => {
-    setChatHistory(chatHistory);
-  }, [chatHistory]);
+    if(isChatBoxVisible)
+      chatBox.current.scrollTop = chatBox.current.scrollHeight;
+  }, [chatHistory, isChatBoxVisible]);
 
-  useEffect(
-    (event) => {
-      if (isChatBoxVisible)
-        chatBox.current.scrollTop = chatBox.current.scrollHeight;
-    },
-    [chatHistory, isChatBoxVisible]
-  );
+  const newMessage = () => {
+
+    let users = [
+      'a',
+      'b',
+    ];
+
+    let randomUser = Math.floor(Math.random()*2);
+
+    return {
+      id: Math.random().toString(32),
+      user: users[randomUser],
+      message: message,
+    }
 
+  }
+  
   const onClickToggleChatBox = () => setIsChatBoxVisible(!isChatBoxVisible);
 
-  const onClickAddMessage = () => {};
+  const onClickAddMessage = () => {
+    if(message.trim().length===0) return;
+    setChatHistory([...chatHistory, newMessage()]); 
+    setMessage('');
+  };
 
-  const onChangeMessage = (event) => {};
+  const onChangeMessage = (event) => {
+    setMessage(event.target.value);
+  };
+
+  const onKeyUpInput = (event) => {
+    const { keyCode } = event;
+    if(keyCode === 13 && event.ctrlKey) {
+      onClickAddMessage();
+    }
+  }
 
   return (
-    <div className="webchat-container">
-      <div
-        className={`webchat-panel-inner webchat-toggle--${
-          isChatBoxVisible ? "show" : "hide"
-        }`}
-      >
-        <div className="webchat-bar">
-          <p className="webchat-bar-spacing">How may we help you?</p>
+    <div className="lqe-wct">
+      <div className={`lqe-wct__main lqe-wct__main__toggle--${isChatBoxVisible ? "show" : "hide"}`}>
+        <div className="lqe-wct__main__hdr">
+          <p className="lqe-wct__main__hdr_text_s">How may I help you?
```

Comment:
As discussed should this be lqe-wct__main__hdr__text__s to indicate that web chat has a main that has a header that has text that has spacing???

Reply:
I think Xin said position and spacing is one underscore.. but if that is the case its still wrong it should be __text_s